game-core/src/main/java/games/strategy/triplea/delegate/StrategicBombingRaidBattle.java

```diff
@@ -315,33 +316,34 @@ private void showBattle(final IDelegateBridge bridge) {
     getDisplay(bridge).showBattle(battleId, battleSite, title, attackingUnits, defendingUnits, null, null, null,
         Collections.emptyMap(), attacker, defender, isAmphibious(), getBattleType(),
         Collections.emptySet());
-    getDisplay(bridge).listBattleSteps(battleId, m_steps);
+    getDisplay(bridge).listBattleSteps(battleId, steps);
   }
 
-  class FireAA implements IExecutable {
+  class FireAa implements IExecutable {
     private static final long serialVersionUID = -4667856856747597406L;
```

Comment:
I wonder if it would work to convert those anonymous classes to lambdas while still being able to deserialize them.
I tried to find some information on this topic and while it seems to be possible I learned something new about java.
I don't know if you knew that already, but [it's possible to cast a lambda expression to multiple types during creation](https://stackoverflow.com/a/22808112/5769952), also generic type specifications can also have a class "extend" multiple interfaces (and one class) this way 😱

Reply:
I had a quick glance over the implementations of IExecutable.
Most of the implementations are anonymous classes, there exist 7 (I think +/-2) named classes that in most cases implement the method.
There seem to be a lot of IExecutables adding other IExecutables to the stack, which internally mutates the state of the outer IExecutable for the inner IExecutables to proceed with that internal state.
I'm pretty sure most `IExecutable`s look like this, and the ugly mutability can be avoided by nesting `IExecutable`s even further into each other, but given the amount of anonymous classes (about 60), this would be a rather huge task to do, but I'm convinced it could do a lot in terms of readability.
We could even go as far as replacing `IExecutable `with a `BiConsumer` and just require the implementation in `IExecutionStack `to implement serializeable as well. Or If we're really crazy use ´BiFunction` and modify `IExecutionStack` store the last execution result to be used in the next function or something, so we have a structure that kinda reminds me of javascript async waterfalls, which keeps the nesting rather flat regardless of the callbacks. But yeah, probably overkill anyways.